homeassistant/components/lock/__init__.py

```diff
@@ -110,3 +127,38 @@ def state(self):
         if locked is None:
             return STATE_UNKNOWN
         return STATE_LOCKED if locked else STATE_UNLOCKED
+
+class CodeLockControlPanel(Entity):
```

Comment:
The added functionality should be part of the `LockDevice`. There should not be an extra class.